This patch removes the outdated test logic for ant/gradle, and forces gradle build
--- opencv-4.0.1/cmake/OpenCVDetectPython.cmake.orig	2018-12-22 08:03:30.000000000 +0100
+++ opencv-4.0.1/cmake/OpenCVDetectPython.cmake	2019-01-31 11:33:10.896502978 +0100
--- OpenCVDetectAndroidSDK.cmake	2020-08-19 23:04:26.004962281 +0200
+++ OpenCVDetectAndroidSDK.cmake.new	2020-08-19 23:06:40.894583256 +0200
@@ -173,43 +173,11 @@
   ocv_detect_android_sdk_tools()
   ocv_detect_android_sdk_build_tools()
 
-  if(ANDROID_SDK_TOOLS_VERSION VERSION_LESS 14)
-    message(FATAL_ERROR "Android SDK Tools: OpenCV requires Android SDK Tools revision 14 or newer.\n"
-                        "${__msg_BUILD_ANDROID_PROJECTS}")
-  endif()
 
-  if(NOT ANDROID_SDK_TOOLS_VERSION VERSION_LESS 25.3.0)
-    message(STATUS "Android SDK Tools: Ant (Eclipse) builds are NOT supported by Android SDK")
-    ocv_update(ANDROID_PROJECTS_SUPPORT_ANT OFF)
-    if(NOT ANDROID_SDK_BUILD_TOOLS_VERSION VERSION_LESS 26.0.2)
-      # https://developer.android.com/studio/releases/gradle-plugin.html
-      message(STATUS "Android SDK Build Tools: Gradle 3.0.0+ builds support is available")
-      ocv_update(ANDROID_PROJECTS_SUPPORT_GRADLE ON)
-    endif()
-  else()
-    include(${CMAKE_CURRENT_LIST_DIR}/../OpenCVDetectApacheAnt.cmake)
-    if(ANT_EXECUTABLE AND NOT ANT_VERSION VERSION_LESS 1.7)
-      message(STATUS "Android SDK Tools: Ant (Eclipse) builds are supported")
-      ocv_update(ANDROID_PROJECTS_SUPPORT_ANT ON)
-    endif()
-  endif()
+  ocv_update(ANDROID_PROJECTS_SUPPORT_GRADLE ON)
+  ocv_update(ANDROID_PROJECTS_BUILD_TYPE "GRADLE")
 
-  if(NOT DEFINED ANDROID_PROJECTS_BUILD_TYPE)
-    if(ANDROID_PROJECTS_SUPPORT_ANT)
-      ocv_update(ANDROID_PROJECTS_BUILD_TYPE "ANT")
-    elseif(ANDROID_PROJECTS_SUPPORT_GRADLE)
-      ocv_update(ANDROID_PROJECTS_BUILD_TYPE "GRADLE")
-    else()
-      message(FATAL_ERROR "Android SDK: Can't build Android projects as requested by BUILD_ANDROID_PROJECTS=ON variable.\n"
-                          "${__msg_BUILD_ANDROID_PROJECTS}")
-    endif()
-  endif()
-
-  if(ANDROID_PROJECTS_BUILD_TYPE STREQUAL "ANT")
-    message(STATUS "Android SDK Tools: Prepare Android projects for using Ant build scripts (deprecated)")
-  elseif(ANDROID_PROJECTS_BUILD_TYPE STREQUAL "GRADLE")
-    message(STATUS "Android SDK Tools: Prepare Android projects for using Gradle 3.0.0+ build scripts")
-  endif()
+  message(STATUS "Android SDK Tools: Prepare Android projects for using Gradle 3.0.0+ build scripts")
 
 else()
   message("Android: Projects builds are DISABLED")
